mockbroker_test.go

```diff
@@ -77,6 +84,10 @@ func (b *mockBroker) BrokerID() int32 {
 	return b.brokerID
 }
 
+func (b *mockBroker) History() []RequestResponse {
+	return b.history
```

Comment:
The workflow that I expected and in fact use it in our offset manager implementation to inspect the history at the very end of the test when there are no more modifications. But it does make sense to make this method more universal and indeed return a copy. 

And you are right about the lock, for visibility guarantee if for nothing else.